src/GraphQL/Types/GraphTypesLookup.cs

```diff
@@ -329,7 +339,10 @@ private void HandleField(Type parentType, FieldType field, TypeCollectionContext
 
             foreach (var arg in field.Arguments)
             {
-                arg.Name = FieldNameConverter.NameFor(arg.Name, null);
+                if (parentType != typeof(SchemaIntrospection))
+                {
+                    arg.Name = FieldNameConverter.NameFor(arg.Name, null);
```

Comment:
I want to once again focus your attention on this call (and a similar call in one more place). This method for fields is used for the argument. **An argument is not a field**. I believe that these 2 places need to be changed anyway.

Reply:
We need to think about whether argument names should be converted at all. Possible not. Related to #1081

Reply:
@pavel-usachev What guided you when you thought that argument names should also be converted? I see that you specifically added tests where you call `GetArgument` with different arg name. Why it was impossible for you to get the value of the argument by its correct name?

Reply:
In the referenced PR #1081 , the change makes it so that GetArgument converts argument names so they match the already-converted argument name (by `HandleField`) and the value can be returned.  I experienced this issue myself with an older release.

As an example, if you define an argument `MyArg`, then `context.GetArgument("MyArg")` should return the proper value, whether or not the argument name had been converted to camel case (i.e. `myArg`).  This would fail prior to #1081 .

Personally, I have no issue with argument names not being converted.  Then we would remove the applicable code from `HandleField` and revert #1081 .  In the prior example, `MyArg` would simply stay `MyArg`.

Be aware that changing this functionality would cause significant breaking changes in existing user codebases, if the user had used pascal case argument names in their code but are expecting camel case argument names exposed through graphql.

Reply:
OK. I see now. You just adjusted to the argument names obtained when constructing the schema. So...
@Shane32 I see two three options:
1. Make new converter interface or method for converting argument names, with or without additional check for introspection args. I think that **with**.
2. Remove two argument conversion calls and not provide ability to change argument names anyway.
3. Your option with `if` check only, `NameFor` does not change.

Honestly, I would prefer 1 or 2 because I don't like that:
```c#
 // do not use context.GetArgument<> as it will translate the argument name via the selected schema's IFieldNameConverter
 var includeDeprecated = (bool)context.Arguments["includeDeprecated"];
```

Reply:
I like option 2 the best, but am concerned about it breaking existing user code. Thoughts ?

Reply:
After discussion with @pavel-usachev  we came to the conclusion that following option 1 would be better - there will be no hidden breaking changes. And besides, there is nothing wrong with enabling the conversion of argument names, we just need the appropriate API for this.

Reply:
> Since all of the introspection types are part of the internal graphql library, I think it makes sense to remove the "is this an introspection type?" question from the realm of the IFieldNameConverter altogether, and put the burden on the graphql library to call the correct field name converter for introspection fields.

Agree.

> Be aware that changing this functionality would cause significant breaking changes in existing user codebases, if the user had used pascal case argument names in their code but are expecting camel case argument names exposed through graphql.

Also agree. So option 1 with check.

> Or, we could expand upon the functionality by changing NameFor to NameForField and adding NameForArgument, and passing the proper type in when calling NameForArgument. This raises the question: should NameForArgument have a third parameter to indicate which field? This surely would provide the most flexibility for implementers.

I would like to see two methods, one for fields and one for args, within one interface (`IFieldNameConverter` -> `INameConverter`). It is a breaking change for custom converters (which no one uses I think) but explicit and simple to implement.

Reply:
I like that answer and will update the PR

Reply:
The PR has been updated with INameConverter and two methods NameForField and NameForArgument